app/src/main-process/app-window.ts

```diff
@@ -324,6 +325,22 @@ export class AppWindow {
     }
   }
 
+  /** Handle when a modal dialog is opened. */
+  public dialogOpened() {
+    if (this.window.isFocused()) {
+      // No additional notifications are needed.
+      return
+    }
+    if (__DARWIN__) {
+      shell.beep()
+      // This is used instead of flashFrame in order to match the OS dialog behavior.
+      app.dock.bounce('critical')
+    } else {
+      this.window.once('focus', () => this.window.flashFrame(false))
+      this.window.flashFrame(true)
+    }
```

Comment:
Could you add a comment explaining this stuff? Basically, why beeping on macOS and using `bounce('critical')` vs `flashFrame` on Windows (feel free to link any of the UX docs you provided in this PR if needed)